### PR TITLE
add coerce option for text and numbers types

### PR DESCRIPTION
**Pull request by PSeitz**
allow to coerce the field type when indexing if the type does not match

### CODE REVIEW

File: src/schema/field_type.rs [hunk 1]
```diff
@@ -329,11 +329,50 @@ impl FieldType {
...
+                        } else {
+                            Err(ValueParsingError::TypeError {
+                                expected: "a u64",
+                                json: JsonValue::String(field_text),
+                            })
+                        }
+                    }
+                    FieldType::I64(opt) => {
+                        if opt.should_coerce() {
+                            Ok(Value::U64(field_text.parse().map_err(|_| {
```
Comment (fulmicoton) [inline src/schema/field_type.rs]:
```suggestion
                            Ok(Value::I64(field_text.parse().map_err(|_| {
```

File: src/schema/field_type.rs [hunk 2]
```diff
@@ -329,11 +329,50 @@ impl FieldType {
...
+                        } else {
+                            Err(ValueParsingError::TypeError {
+                                expected: "a i64",
+                                json: JsonValue::String(field_text),
+                            })
+                        }
+                    }
+                    FieldType::F64(opt) => {
+                        if opt.should_coerce() {
+                            Ok(Value::U64(field_text.parse().map_err(|_| {
```
Comment (fulmicoton) [inline src/schema/field_type.rs]:
```suggestion
                            Ok(Value::F64(field_text.parse().map_err(|_| {
```

File: src/schema/field_type.rs [hunk 3]
```diff
@@ -329,11 +329,50 @@ impl FieldType {
                         Ok(DateTime::from_utc(dt_with_fixed_tz).into())
                     }
                     FieldType::Str(_) => Ok(Value::Str(field_text)),
-                    FieldType::U64(_) | FieldType::I64(_) | FieldType::F64(_) => {
-                        Err(ValueParsingError::TypeError {
-                            expected: "an integer",
-                            json: JsonValue::String(field_text),
-                        })
+                    FieldType::U64(opt) => {
+                        if opt.should_coerce() {
```
Comment (fulmicoton) [inline src/schema/field_type.rs:333]: We need unit tests for this.

**Review by fulmicoton — CHANGES_REQUESTED** [review]
Unit tests needed (that's a good change though)

**Comment by PSeitz** [comment]
> Unit tests needed (that's a good change though)

Yes, I was just checking first if that's the right direction

File: src/schema/numeric_options.rs [hunk 4]
```diff
@@ -17,6 +18,12 @@ pub struct NumericOptions {
     fieldnorms: bool, // This attribute only has an effect if indexed is true.
     fast: bool,
     stored: bool,
+    #[serde(skip_serializing_if = "is_false")]
```
Comment (fulmicoton) [inline src/schema/numeric_options.rs:21]: is there a reason to skip serialization? 
```suggestion
```

File: src/schema/text_options.rs [hunk 5]
```diff
@@ -17,6 +17,14 @@ pub struct TextOptions {
     stored: bool,
     #[serde(default)]
     fast: bool,
+    #[serde(default)]
+    #[serde(skip_serializing_if = "is_false")]
```
Comment (fulmicoton) [inline src/schema/text_options.rs:21]:
```suggestion
```

**Review by fulmicoton — APPROVED** [review]

File: src/schema/numeric_options.rs (hunk 4 above)
Comment (PSeitz) [reply src/schema/numeric_options.rs:21]: just a little less noise when reading the meta.json

**Pull request merged**